portable/CMakeLists.txt

```diff
@@ -0,0 +1,276 @@
+# FreeRTOS internal cmake file. Do not use it in user top-level project
+
+add_library(freertos_kernel_port STATIC
```

Comment:
Could the port be specified as a separate static library?

Reply:
External (separate) port is kind of advanced feature which is not so requested.
Maybe let's start with existing functionality and add such features later in scope of separate PR?

Reply:
I am inclined to agree with @yhsb2k. We can move stuff into new files/modularize this; change the cmake to add other libraries etc - but that can be done incrementally. This PR seems like a good starting point. What do you think @paulbartell?